### PR TITLE
[textual_inversion_sdxl.py] fix lr scheduler steps count

### PR DESCRIPTION
**Pull request by ca1yz**
# What does this PR do?

Part of #8384
Fix for LR in a distributed training when num_train_epoch is passed

run with the following scripts:

```python
from huggingface_hub import snapshot_download

local_dir = "./cat"
snapshot_download(
    "diffusers/cat_toy_example", local_dir=local_dir, repo_type="dataset", ignore_patterns=".gitattributes"
)

```

```bash
#!/bin/bash

mkdir -p textual_inversion_cat/logs

accelerate launch \
  --multi_gpu \
  --num_processes=2 \
  --mixed_precision="bf16" \
  textual_inversion_sdxl.py \
  --pretrained_model_name_or_path="stabilityai/stable-diffusion-xl-base-1.0" \
  --train_data_dir="./cat" \
  --output_dir="textual_inversion_cat" \
  --placeholder_token="<my-token>" \
  --initializer_token="cat" \
  --resolution=1024 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=1 \
  --max_train_steps=100 \
  --learning_rate=1e-4 \
  --num_vectors=4 \
  --validation_prompt="a photo of a <my-token>" \
  --num_validation_images=4 \
  --validation_steps=100 \
  --save_steps=500 \
  --mixed_precision="bf16" \
  --enable_xformers_memory_efficient_attention \
  --gradient_checkpointing \
  --center_crop \
  --report_to="tensorboard" \
  --logging_dir="textual_inversion_cat/logs"
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sayakpaul

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Thank you.
